src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -94,22 +131,32 @@ public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone)
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return OffsetDateTime.of(localTime, ZoneOffset.of(String.valueOf(OFFSET)));
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.parse(date, DateTimeFormatter.BASIC_ISO_DATE));
+        } catch (DateTimeException e) {
+            return Optional.empty();
+        }
     }
 
     /**
      * Given a String formatted as `d MMM yyyy` (MMM - Sep, Oct, etc),
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> customParseDate(String date) {
+        try {
+            LocalDate dateTime = LocalDate.parse(date, DATE_FORMATTER);
+            return Optional.of(dateTime);
+        } catch (DateTimeException e) {
+            e.printStackTrace();
```

Comment:
Same

Reply:
I disagree with you, what exactly is "bad practice"?